### DISCUSSION TITLE
How to initilize thrust device vector from another device vector declared by cuda

### DISCUSSION THREAD

**Discussion by ztdepztdep** [NVIDIA/cccl#1038] [Thrust] [answered]
I have a device vector d_b[100], I want to creat a thrust vector with d_b[80:100]. How to resolve this mission.

**Comment by miscco** [comment]
I am not fully sure what you mean with `d_b[80:100]`

However, you could always use one of the algorithms such like https://github.com/NVIDIA/thrust/blob/main/thrust/fill.h

**Comment by ztdepztdep** [comment]
> I am not fully sure what you mean with `d_b[80:100]`
> 
> However, you could always use one of the algorithms such like https://github.com/NVIDIA/thrust/blob/main/thrust/fill.h

I mean fill thrust vector with  d_b from the 80'th to the 100'th values.

**Comment by miscco** [comment]
In that case you would need to use `thrust::copy` or `thrust::copy_n` You can find some documentation here https://github.com/NVIDIA/thrust/blob/125d2812f8b68ee38b35d9beb2bf6423a26a1880/thrust/copy.h#L118-L130

**Comment by pauleonix** [comment, marked as answer]
If possibe, I would rather just use the right constructor to avoid unnecessary initialisation, i.e.
```c++
thrust::device_vector<float> d_c(d_b.cbegin() + 80, d_b.cend());
// or
thrust::device_vector<float> d_c(d_b.cbegin() + 80, d_b.cbegin() + 100);
```

**Comment by Pavelevich** [comment]
this can work for u:
* device vector is already with some values

int main()
{
    int size = 299;
    int d_b[299];
    thrust::device_vector<int> d(d_b, d_b + size);
    thrust::device_ptr<int> d_ptr = thrust::make_device_ptr(d_b);
    thrust::device_vector<int> d_subset(d_ptr + 66, d_ptr + 299);
    return 0;
}
